be/src/olap/page_cache.h

```diff
@@ -28,11 +28,62 @@
 
 #include "olap/lru_cache.h"
 #include "util/slice.h"
-
+#include "vec/common/allocator.h"
+#include "vec/common/allocator_fwd.h"
 namespace doris {
 
 class PageCacheHandle;
 
+template <typename TAllocator>
+class PageBase : private TAllocator {
+public:
+    PageBase() : size(0), bytes(0) {}
+
+    PageBase(const char* d, size_t n, size_t b) : data(const_cast<char*>(d)), size(n), bytes(b) {}
+
+    PageBase(const PageBase& other) : data(other.data), size(other.size), bytes(other.bytes) {}
+
+    PageBase(PageBase&& other) : data(other.data), size(other.size) {
+        std::swap(bytes,
+                  other.bytes); // other will be read-only, will not free data and reset_size.
+    }
+
+    PageBase(size_t b) : size(b), bytes(b) {
+        data = reinterpret_cast<char*>(TAllocator::alloc(b, ALLOCATOR_ALIGNMENT_16));
+    }
+
+    ~PageBase() {
+        if (bytes > 0) {
+            TAllocator::free(data, size);
+        }
+    }
+
+    void reset_size(size_t n) {
+        DCHECK(n <= bytes);
+        size = n;
+    }
+
+    void swap(PageBase& p) {
+        std::swap(data, p.data);
+        std::swap(size, p.size);
+        std::swap(bytes, p.bytes);
+    }
+
+    void release() {
+        data = nullptr;
+        size = 0;
+        bytes = 0;
+    }
+
+    char* data;
+    // Effective size, smaller than bytes, such as data page remove checksum suffix.
+    size_t size;
```

Comment:
change name to  charge and size, same as LRUCache's definition.

Reply:
The `size` here is different from LRUCache's `size`. LRUCache's `size = charge + handle_size` is not the difference between the effective size and the actual memory size.

And charge seems to be a concept in lru cache, and page bytes seems to be easier to understand when describing actual memory?